### ISSUE TITLE
Github Action Failed - Proxy not supported

### ISSUE THREAD

**Issue by mohammedazfar-sanuk** [daspn/private-actions-checkout#28]
Hi All,

Github Action is failing when we trying use proxy, below are the error.( eventhough we use http_proxy https_proxy environment values ) Any suggestions pl

App > Cloning using GitHub App strategy
App > Base 64 detected on private key: true
(node:15) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
App > Installation: 1000000
Error: HttpError: request to https://api.github.com/app/installations/1000000/access_tokens failed, reason: connect ETIMEDOUT 140.82.1[21](https://github.com/repos-uk/service/runs/10909091?check_suite_focus=true#step:3:21).5:443
Error: request to https://api.github.com/app/installations/1576859/access_tokens failed, reason: connect ETIMEDOUT 140.82.121.5:443
Error: App > App token generation failed. Workflow can not continue

**Comment by mohammedazfar-sanuk** [comment]
@daspn  @gwatts @droidpl - pl share your suggestions

**Comment by gwatts** [comment]
would of thought setting `http_proxy` as an environment variable in the workflow would work, but i hadn't tested that.. will try to reproduce when i have a few minutes

**Comment by mohammedazfar-sanuk** [comment]
> would of thought setting `http_proxy` as an environment variable in the workflow would work, but i hadn't tested that.. will try to reproduce when i have a few minutes

Thanks @gwatts , let me give a try

**Comment by mohammedazfar-sanuk** [comment]
@gwatts - i have tried with http_proxy, but no luck. pl ref below..

Run daspn/private-actions-checkout@v2
  with:
    actions_list: ["application-action@main"]
    checkout_base_path: ./.github/actions
    app_id: 100009
    app_private_key: ***
  env:
    HTTP_PROXY: http://internal-proxy:80
    HTTPS_PROXY: http://internal-proxy:80
    NO_PROXY: localhost,127.0.0.1,169.254.169.254
    http_proxy: http://internal-proxy:80
    https_proxy: http://internal-proxy:80
    no_proxy: localhost,127.0.0.1,169.254.169.254
/usr/bin/docker exec  3f1f9946a66d22ef6e27cd08b71d7a8c3c42e836b0bf63db18a514257c627305 sh -c "cat /etc/*release | grep ^ID"
App > Cloning using GitHub App strategy
App > Base 64 detected on private key: true
(node:16) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
App > Installation: 100009
Error: HttpError: request to https://api.github.com/app/installations/100009/access_tokens failed, reason: connect ETIMEDOUT 140.82.121.6:443
Error: request to https://api.github.com/app/installations/100009/access_tokens failed, reason: connect ETIMEDOUT 140.82.121.6:443
Error: App > App token generation failed. Workflow can not continue

**Comment by droidpl** [comment]
I think this may be related to the fact that we are using here `@octokit/app` and not the `@octokit/action` which does support the setting. Probably adding the `http_proxy` library to the corresponding request would work it would require some development to see how to integrate it. I don't have push permissions here so @daspn should be the one to handle that 😄

**Comment by daspn** [comment]
@droidpl I just sent an invite so you can contribute directly. I'm really busy these days, so I won't have the opportunity to take a closer look before the weekend!

**Comment by mohammedazfar-sanuk** [comment]
Thanks @droidpl @daspn. pl let me know once you pushed the changes.

**Comment by mohammedazfar-sanuk** [comment]
@daspn  - i have tried with latest changes, but still the issue is not resolved.

Run daspn/private-actions-checkout@master
  with:
    actions_list: [""application-action@main""]
    checkout_base_path: ./.github/actions
    app_id: 100009
    app_private_key: ***
  env:
    https_proxy: http://internal-proxy:80
/usr/bin/docker exec  7b5039695469589[2](f35c sh -c "cat /etc/*release | grep ^ID"
App > Cloning using GitHub App strategy
App > Base 64 detected on private key: true
(node:113) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
App > Installation: 15[8](https://github.com/private/service/runs/5213289736?check_suite_focus=true#step:4:8)85[9](https://github.com/private/service/runs/5213289736?check_suite_focus=true#step:4:9)38
Error: HttpError: request to https://api.github.com/app/installations/100009/access_tokens failed, reason: connect ETIMEDOUT 140.82.[12](https://github.com/private/service/runs/5213289736?check_suite_focus=true#step:4:12)1.6:443
Error: request to https://api.github.com/app/installations/100009/access_tokens failed, reason: connect ETIMEDOUT [14](https://github.com/private/service/runs/5213289736?check_suite_focus=true#step:4:14)0.82.121.6:443
Error: App > App token generation failed. Workflow can not continue

**Comment by daspn** [comment]
Hi @mohammedazfar-sanuk ,

The changes I made (on the master branch) were just security updates suggested by Dependabot and small changes to fix the tests (removed Ubuntu 16 - not supported anymore and added Ubuntu 20).

I took a quick look on what needs to be done as suggested by @droidpl and it's more complex than I thought (a completely different API). I will need either support from the community or more time to work on it, but I'm really busy lately.

Best,
